package/network/utils/wireguard-tools/files/wireguard.init

```diff
@@ -0,0 +1,122 @@
+#!/bin/sh /etc/rc.common
+
+START=80
+USE_PROCD=1
+
+WG_DIR="/tmp/wireguard"
+
+wireguard_check_peer(){
+	local cfg="${1}"
+	local cfile="${2}"
+
+	uci show "network.${cfg}" >> "${cfile}"
+}
+
+wireguard_check_interface() {
+	local cfg="${1}"
+	local proto cfile n_sum o_sum
+
+	config_get proto "${cfg}" proto
+	[ "${proto}" = "wireguard" ] || return 0
+	cfile="$(mktemp -p "${WG_DIR}")"
+	config_foreach wireguard_check_peer "wireguard_${1}" "${cfile}"
+
+	. /lib/functions/network.sh
+
+	n_sum="$(md5sum "${cfile}" | cut -d" " -f1)"
+	rm -rf "${cfile}"
+	[ -f "${WG_DIR}/${cfg}.check" ] && {
+		o_sum="$(cat "${WG_DIR}/${cfg}.check")"
+		[ "${o_sum}" != "${n_sum}" ] && {
+			network_is_up "${cfg}" && ifup "${cfg}"
+		}
+	}
+	echo "$n_sum" > "${WG_DIR}/${cfg}.check"
+}
+
+wireguard_fw_rules() {
+	local network="$1"
+	local port="$2"
+
+	local zone device
+
+	zone="$(fw3 -q network "$network" 2>/dev/null)"
+
+	network_get_device device "$network"
+
+	[ -n "$device" ] && {
+		json_add_object ""
+		json_add_string type rule
+		json_add_string src "$zone"
+		json_add_string device "$device"
+		json_add_string proto udp
+		json_add_string dest_port "$port"
+		json_add_string target ACCEPT
+		json_close_object
+	}
+}
+
+wireguard_start_interface() {
+	local cfg="${1}"
+
+	local proto listen_port zone
+
+	config_get proto "${cfg}" proto
+	[ "${proto}" = "wireguard" ] || return 0
+
+	config_get listen_port "${cfg}" listen_port
+
+	procd_open_instance "${cfg}"
+	procd_set_param command /bin/true
+	procd_set_param stdout 1
+	procd_set_param stderr 1
+	[ -n "$listen_port" ] && {
+		procd_open_data
+		json_add_array firewall
+		config_list_foreach "$cfg" allowed_network wireguard_fw_rules "$listen_port"
+		json_close_array
+		procd_close_data
+	}
+	procd_close_instance
+}
+
+service_started() {
+	procd_set_config_changed firewall
+}
+
+service_stopped() {
+	procd_set_config_changed firewall
+}
+
+service_triggers() {
+	procd_add_reload_trigger "network"
+}
+
+start_service() {
+	mkdir -p "${WG_DIR}"
+	config_load network
+	config_foreach wireguard_check_interface interface
+
+	config_foreach wireguard_start_interface interface
+
+	# delete old peers of related wireguard interface
+	wireguard_check_peers
+}
+
+wireguard_check_peers() {
+	local iface peer
+
+	# get all wireguard peers
+	for peer in $(uci show network | grep =wireguard_); do
+		# extract peer section type
+		peer="${peer##*=}"
+		# extract interface name
+		iface="${peer#*_}"
+
+		# delete peer if iface is not present anymore
+		if ! uci -q show "network.${iface}" 1>/dev/null 2>/dev/null; then
+			uci -q delete "network.@${peer}[-1]"
```

Comment:
I still think that is is a no-go to modify (and commit!) network configuration from an init script. This has unpredictable side effects, might accidentally apply intentionally uncommitted changes, destroy formatting and comments in the configuration etc.

Reply:
Now that you mention it! I dimly remember you commenting on the mailing list before. But then how am I supposed to do that? The way I see it, it'll have to happen in the LuCI.

Reply:
This Is done now in LuCI
openwrt-19.07:
https://github.com/openwrt/luci/commit/86d0bf86eabd04ba397a7e20ea84f892be8ec934

master:
https://github.com/openwrt/luci/commit/890dcac2de7b1d6bc6768c69ea7129d27a809ef9